core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,225 @@
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
+    val keySize = CompactSizeUInt.parse(bytes)
+    val key = bytes.drop(keySize.size).take(keySize.toLong)
+    val valueSize =
+      CompactSizeUInt.parse(bytes.drop(keySize.size + keySize.toLong))
+    val value = bytes
+      .drop(keySize.size + keySize.toLong + valueSize.size)
+      .take(valueSize.toLong)
+    val totalSize = keySize.size + keySize.toLong + valueSize.size + valueSize
+      .toLong
+    (totalSize, key, value)
+  }
+
+  @tailrec
+  final def parseGlobalMap(
+      remainingBytes: ByteVector,
+      accum: Seq[GlobalPSBTRecord],
+      tx: Transaction): (ByteVector, Seq[GlobalPSBTRecord], Transaction) = {
+    if (remainingBytes.head == 0x00.byteValue) // terminator
+      (remainingBytes.tail, accum, tx)
+
+    val (totalSize, key, value) = parseKeyAndValue(remainingBytes)
+    val rest = remainingBytes.drop(totalSize) //Drop the key-value map
+    require(
+      rest.head == 0x00.byteValue,
+      s"Key-value maps must be separated by the separator char 0x00 was: ${ByteVector(rest.head)}")
+    val next = rest.tail
+    PSBTGlobalKey.fromByte(key.head) match {
+      case UnsignedTransactionKey =>
+        println("DEBUGGING HERE")
+        println(key)
+        println(value)
+        println(next)
+        if (tx != null)
+          throw new Exception("Duplicate Key, unsigned tx already provided")
+
+        require(key.size == 1, "The key must only contain the 1 byte type")
+
+        val transaction: Transaction = Transaction.fromBytes(value)
+        parseGlobalMap(next,
+                       accum :+ UnsignedTransaction(transaction),
+                       transaction)
+      case XPubKeyKey =>
+        val xpub = ExtPublicKey.fromBytes(key.tail.take(78))
+        val fingerprint = value.take(4)
+        val path = BIP32Path.fromBytes(value.drop(4))
+        parseGlobalMap(next, accum :+ XPubKey(xpub, fingerprint, path), tx)
+      case VersionKey =>
+        require(key.size == 1, "The key must only contain the 1 byte type")
+
+        val version = UInt32(value)
+        parseGlobalMap(next, accum :+ Version(version), tx)
+    }
+  }
+
+  @tailrec
+  def parseInputMaps(
+      remainingBytes: ByteVector,
+      remainingInputs: Int,
+      accum: Seq[InputPSBTMap]): (ByteVector, Seq[InputPSBTMap]) = {
+    if (remainingInputs <= 0)
+      return (remainingBytes, accum)
```

Comment:
is general `return` isn't needed in scala. You need it here because you aren't using `{}` around your `if () { } else { }` expression. 

This should fix your problem 

```scala
    if (remainingInputs <= 0) {
      (remainingBytes, accum)
    } else {
      val (next, map) = PSBTHelper.parseInputMap(remainingBytes, Nil)

      require(next.head == 0xff.byteValue,
        "Input maps should be separated by 0xff")

      parseInputMaps(next.tail,
        remainingInputs - 1,
        accum :+ InputPSBTMap(map.toVector))      
    }
```